### PR TITLE
add mailroom.py

### PR DESCRIPTION
**Pull request by heb2016**
add mailroom.py

### CODE REVIEW

**Comment by PythonCHB** [comment]
you seem to have a conflict with one of your own files. -- you can follow the gitHub instructions, or, on your laptop:

git pull upstream master

(you may get a conflict -- you can then resolve that and "git commit")

git push

and it should get cleaned up.

I won't merge 'till the conflict is respolved.

**Comment by PythonCHB** [comment]
hmm -- there are a LOT of changed files, not all in your own dir. When you make a PR, make sure to look and see what's changed.

Also, when you do a commit, make sure you are committing what you intend before finalizing it.

Any chance you are on Windows, and looked at these files in an editor that changed teh line endings? Just guessing here...

but I'm not going to merge all these changes, so we'll need to figure out how to clean this up first.

meanwhile, I'll look at your code...

File: students/heb2016/session02/series.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+
+## Session02 Exercise 
+# fibonacci 
+def fibonacci(n):
+		row=0
```
Comment (PythonCHB) [inline students/heb2016/session02/series.py]: only use 4 spaces for indentation -- always!

Comment (PythonCHB) [inline students/heb2016/session02/series.py]: hmm -- this is the only file I see in the heb2016 dir... is there anything else?

**Comment by PythonCHB** [comment]
OK -- this is way too much of a mess.

Maybe you have git set up to change line endings? so it did it for a whole bunch of other files?

Anyway, I'm going to close this and we can figure out how to clean it up on Tuesday if need be.

**Comment by heb2016** [comment]
I did 'got pull' & 'git push' on MAC, and these is a screenshot:

I have committed (pull request as well) two programs so far: mailroom.py,
robt13.py


sea-heb-m:IntroPython2016 heb$ git pull upstream master

remote: Counting objects: 34, done.

remote: Compressing objects: 100% (8/8), done.

remote: Total 34 (delta 8), reused 7 (delta 7), pack-reused 19

Unpacking objects: 100% (34/34), done.

From https://github.com/UWPCE-PythonCert/IntroPython2016

 * branch            master     -> FETCH_HEAD

   7fe1616..d5be645  master     -> upstream/master

subl -n -w: subl: command not found

error: There was a problem with the editor 'subl -n -w'.

Not committing merge; use 'git commit' to complete the merge.

sea-heb-m:IntroPython2016 heb$ git push

warning: push.default is unset; its implicit value is changing in

Git 2.0 from 'matching' to 'simple'. To squelch this message

and maintain the current behavior after the default changes, use:


  git config --global push.default matching


To squelch this message and adopt the new behavior now, use:


  git config --global push.default simple


See 'git help config' and search for 'push.default' for further information.

(the 'simple' mode was introduced in Git 1.7.11. Use the similar mode

'current' instead of 'simple' if you sometimes use older versions of Git)


Everything up-to-date

**Comment by heb2016** [comment]
As you see, I got 2 error/alert messages from my MAC git command:
1) something wrong with Sublime text setup, and it got problem to merge
2) Something not perfect, there is a warning message when I push


Beyond this, I got some questions:

1) when this error message mention about 'merge',
   a)  does it mean my local commit/change could not merge to remote master
(my copy version of master on UW server)? or
   b) does it mean my master on the UW server (my copy version of master on
the server) could not merge to the UW version master (on UW server, the
original master of my copy version) ?

2) how this merge works and how my setup prevent me from merging?
How these three steps relate to this merge:
  a) how my commit/push command on local MAC associate to this merge?
  b) how my pull request on UW server (under my own fork) associate to this
merge?

Why you do not see mailroom.py on UW server (under my own git fork branch),
but I indeed have completed these steps: commit/push/pull request


I am very confused at this step...


Thanks for being patience to look at my questions.

Beatrice He

**Comment by heb2016** [comment]
*I got three things to ask in this email:*


*1) Homework mailroom.py--I did adding mailroom.py to git again, here is
the screenshot from my local MAC:*

sea-heb-m:session03 heb$ git add mailroom.py

sea-heb-m:session03 heb$ git commit

subl -n -w: subl: command not found

error: There was a problem with the editor 'subl -n -w'.

Please supply the message using either -m or -F option.

sea-heb-m:session03 heb$ git commit -m 'add mailroom.py'

[master 4cbb0e0] add mailroom.py

sea-heb-m:session03 heb$ git push

warning: push.default is unset; its implicit value is changing in

Git 2.0 from 'matching' to 'simple'. To squelch this message

and maintain the current behavior after the default changes, use:

  git config --global push.default matching

To squelch this message and adopt the new behavior now, use:

  git config --global push.default simple

See 'git help config' and search for 'push.default' for further information.

(the 'simple' mode was introduced in Git 1.7.11. Use the similar mode

'current' instead of 'simple' if you sometimes use older versions of Git)

Counting objects: 52, done.

Delta compression using up to 8 threads.

Compressing objects: 100% (39/39), done.

Writing objects: 100% (40/40), 13.27 KiB | 0 bytes/s, done.

Total 40 (delta 9), reused 0 (delta 0)

remote: Resolving deltas: 100% (9/9), completed with 6 local objects.

To https://github.com/heb2016/IntroPython2016

   c1f15ef..4cbb0e0  master -> master



*2) About returned error messages I did 'git pull' & 'git push' on MAC, and
these is a screenshot:*
I have committed (pull request as well) two programs so far: mailroom.py,
robt13.py

sea-heb-m:IntroPython2016 heb$ git pull upstream master

remote: Counting objects: 34, done.

remote: Compressing objects: 100% (8/8), done.

remote: Total 34 (delta 8), reused 7 (delta 7), pack-reused 19

Unpacking objects: 100% (34/34), done.

From https://github.com/UWPCE-PythonCert/IntroPython2016

 * branch            master     -> FETCH_HEAD

   7fe1616..d5be645  master     -> upstream/master

subl -n -w: subl: command not found

error: There was a problem with the editor 'subl -n -w'.

Not committing merge; use 'git commit' to complete the merge.

sea-heb-m:IntroPython2016 heb$ git push

warning: push.default is unset; its implicit value is changing in

Git 2.0 from 'matching' to 'simple'. To squelch this message

and maintain the current behavior after the default changes, use:

  git config --global push.default matching

To squelch this message and adopt the new behavior now, use:

  git config --global push.default simple

See 'git help config' and search for 'push.default' for further information.

(the 'simple' mode was introduced in Git 1.7.11. Use the similar mode

'current' instead of 'simple' if you sometimes use older versions of Git)

Everything up-to-date


*As you see, I got 2 error/alert messages from my MAC git command:*

1) something wrong with Sublime text setup, and it got problem to merge
2) Something not perfect, there is a warning message when I push





*3)  I got some questions related to 'Merge': *

1) when this error message mention about 'merge',
   a)  does it mean my local commit/change could not merge to remote master
(my copy version of master on UW server)? or
   b) does it mean my master on the UW server (my copy version of master on
the server) could not merge to the UW version master (on UW server, the
original master of my copy version) ?

2) how this merge works and how my setup prevent me from merging?
How these three steps relate to this merge:
  a) how my commit/push command on local MAC associate to this merge?
  b) how my pull request on UW server (under my own fork) associate to this
merge?

Why you do not see mailroom.py on UW server (under my own git fork branch),
but I indeed have completed these steps: commit/push/pull request

I am very confused at this step...





Thanks for being patience to look at my questions.

Beatrice He

**Comment by PythonCHB** [comment]
I see en error in there:

subl -n -w: subl: command not found

error: There was a problem with the editor 'subl -n -w'.

It looks like you tried to configure git to use sublime to edit commit
messages, but that configuration has failed.

Maybe someone in the class has figured this out (I haven't :-) )

I think you can recover by committing manually now:

git commit -a -m "merging with upstream"

-CHB



-- 
Christopher Barker, PhD

Python Language Consulting
  - Teaching
  - Scientific Software Development
  - Desktop GUI and Web Development
  - wxPython, numpy, scipy, Cython

**Comment by heb2016** [comment]
I just did:

sea-heb-m:session03 heb$ git add mailroom.py

sea-heb-m:session03 heb$ git commit -a -m "merging with upstream"

# On branch master

# Untracked files:

#   (use "git add <file>..." to include in what will be committed)

#

# ../1

# ../README.rst

# ../session02/exerc_fizzbuzz.py

# ../session02/exerc_grid_printer.py

# ../session02/grind_printer.py

# .mailroom_test.py.swp

# exer01.py

# list_lab.py

# mailroom_test.py

# ../test.py

# ../test2.py

nothing added to commit but untracked files present (use "git add" to track)

**Comment by heb2016** [comment]
I have re commit , and please elt me know if you are able to merge my files
mailroom.py now

**Comment by PythonCHB** [comment]
Darn! there are a lot of files changed in the PR -- that are not in your students dir.

I'm not sure how this got so messed up :-(

You may need to start fresh -o one of the nice things about git is that you have full copies everywhere...

To start fresh:

1) make sure you have all your own work that you want to save on your own laptop somewhere (could be the git repo).

[this is all the same instructions from the second class:

http://uwpce-pythoncert.github.io/IntroToPython/session02.html#first-a-little-git-primer

2) make a new fork of the class repo, from here:

https://github.com/UWPCE-PythonCert/IntroPython2016

you will need to give it a new name, or maybe have to rename or delete your current fork.

3) clone that fork onto your laptop.
  - copy the https url from gitHub
  - git clone the_url_you_copied
(you may need to rename your existing repo first)

4) set upstream:

git remote add upstream https://github.com/UWPCE-PythonCert/IntroPython2016.git

5) git remote -v to check -- it should look something like:

origin	https://github.com/PythonCHB/IntroPython2016.git (fetch)
origin	https://github.com/PythonCHB/IntroPython2016.git (push)
upstream	https://github.com/UWPCE-PythonCert/IntroPython2016.git (fetch)
upstream	https://github.com/UWPCE-PythonCert/IntroPython2016.git (push)

(but with your github account in the origin one)

6) see if it all works:

git pull upstream master

(that shouldn't do much -- you just got it 00 it should already be in sync)

7) add your files to yoru repo that aren't already there, but copying them to teh repo dirs.

8) git add *.py
you may need to go into a couple session dirs to add them all

9) git commit -a -m "adding my work again!"

10) push to gitHub:

git push origin master

11) got to gitHub and do another PR --hopefully it will only have your files in it.

This seems like a major pain, but sometimes it's easier to start clean than to try to figure out what's wrong...

**Pull request closed without merging**